pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -671,6 +701,70 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	return deployment, nil
 }
 
+func (c *Cluster) createOSDService(osd OSDInfo, labels map[string]string) (string, error) {
+	selectorLabels := map[string]string{
+		k8sutil.AppAttr: AppName,
+		"ceph-osd-id":   strconv.FormatInt(int64(osd.ID), 10),
+	}
+	svcDef := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("rook-ceph-osd-%d", osd.ID),
+			Namespace: c.clusterInfo.Namespace,
+			Labels:    labels,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: selectorLabels,
+			Ports: []v1.ServicePort{
+				{
+					Name:       "osd-port",
+					Port:       6800,
+					TargetPort: intstr.FromInt(6800),
+					Protocol:   v1.ProtocolTCP,
+				},
+				{
+					Name:       "osd-port-2",
+					Port:       6801,
+					TargetPort: intstr.FromInt(6801),
+					Protocol:   v1.ProtocolTCP,
+				},
+			},
+		},
+	}
+	err := c.clusterInfo.OwnerInfo.SetOwnerReference(svcDef)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to set owner reference to osd service %q", svcDef.Name)
+	}
+
+	svc, err := k8sutil.CreateOrUpdateService(c.clusterInfo.Context, c.context.Clientset, c.clusterInfo.Namespace, svcDef)
+	if err != nil {
+		return "", err
+	}
+
+	client, err := mcsv1Client.NewForConfig(c.context.KubeConfig)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get mcs-api client")
+	}
+
+	err = k8sutil.ExportService(c.clusterInfo.Context, client, svc)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to export service")
+	}
+
+	err = k8sutil.VerifyExportedService(c.clusterInfo.Context, client, svc.Name, svc.Namespace)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to verify status of the exported service %q", svc.Name)
+	}
+
+	exportedIP, err := k8sutil.GetExportedServiceIP(c.clusterInfo.Context, c.context.Clientset, svc.Name, svc.Namespace)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to export service for %q", svc.Name)
+	}
+
```

Comment:
This can be moved to the helper as it's repeated in multiple places?

Reply:
updated the code.